### PR TITLE
Resource resize revamp

### PR DESCRIPTION
**Pull request by jridgewell**
Three real changes:
- Small updates to make things more readable
  - Use `Infinity` as the sentinel value
  - Merge the "in viewport" rules
- Ensure `#overflowCallback` is called when the element is above the viewport

### CODE REVIEW

File: src/service/resources-impl.js [hunk 1]
```diff
@@ -850,39 +853,41 @@ export class Resources {
...
           },
           mutate: state => {
-            let minTop = -1;
+            let minTop = Infinity;
+            let diff = 0;
             scrollAdjSet.forEach(request => {
               const box = request.resource.getLayoutBox();
-              minTop = minTop == -1 ? box.top : Math.min(minTop, box.top);
+              minTop = Math.min(minTop, box.top);
+              diff += request.newHeight - box.height;
```
Comment (jridgewell) [inline src/service/resources-impl.js]: Actually, this should only diff if the `box.bottom` is above the current `scrollTop`.

**Comment by dvoytenko** [comment]
@jridgewell Was going over open PRs and found this one. Must have missed it before. Is it still needed?

**Comment by jridgewell** [comment]
I think the last point is still needed.

**Comment by mrjoro** [comment]
Closing since this hasn't been updated in a while; reopen if needed.

**Pull request closed without merging**